src/index.ts

```diff
@@ -48,16 +48,14 @@ export default class ServerlessConventions {
       },
     });
 
-    this.conventionsConfig =
-      this.serverless.service.initialServerlessConfig.conventions;
+    // Make sure `conventionsConfig` is defined in case `serverless.yml` only has empty `conventions` block
+    this.conventionsConfig = this.serverless.service.initialServerlessConfig
+      .conventions || {
+      ignore: {},
+    };
 
-    // Make sure ignore is defined to prevent errors being from being thrown when referencing children
-    if (this.conventionsConfig === undefined) {
-      this.conventionsConfig = {};
-    }
-    if (this.conventionsConfig.ignore === undefined) {
```

Comment:
I like the new syntax more but I'm confused why this check wasn't already handling the issue. Are we sure it's not something else?

Reply:
From what I understand about this plugin, the implementation looks like this:

- `ServerlessConvention` plugin hold a `conventionsConfig` object of type `ConventionsConfig`.
- When we run `serverless conventions-check` command,  `ServerlessConvention` class is construct and it try to copy `this.serverless.service.initialServerlessConfig.conventions` to `conventionsConfig`.
    - If `serverless.yml` does not have `conventions` field -> `this.serverless.service.initialServerlessConfig.conventions` is `undefined` -> `conventionsConfig` is `undefined`.
    - If `serverless.yml` has an empty `conventions` field -> `this.serverless.service.initialServerlessConfig.conventions` is `null` -> `conventionsConfig` is `null` and the old check  `if (this.conventionsConfig === undefined)` failed to catch it.
    - If `serverless.yml` has `conventions` field with an empty `ignore` field -> `this.serverless.service.initialServerlessConfig.conventions.ignore` is `null` -> `conventionsConfig.ignore` is `null` and the check  `if (this.conventionsConfig.ignore === undefined)` failed to catch it.

That's why we got the `TypeError: Cannot read properties of null` error.